src/usecases/shlink-endpoint/add-endpoint.ts

```diff
@@ -0,0 +1,22 @@
+import { SHLinkEndpointModel } from "@/domain/models/shlink-endpoint";
+import { SHLinkEndpointEntity } from "@/entities/shlink-endpoint";
+import { ISHLinkEndpointRepository } from "@/infrastructure/repositories/interfaces/shlink-endpoint-repository";
+import { mapEntityToModel, mapModelToEntity } from "@/mappers/shlink-endpoint-mapper";
+
+export const addEndpointUseCase = async (context: {repo: ISHLinkEndpointRepository}, data: {endpoint: SHLinkEndpointModel}): Promise<SHLinkEndpointModel> => {
+    const oldSHLinkEndpoint = await context.repo.findOne({shlink_id: data.endpoint.getShlinkId()});
```

Comment:
user should not be providing an id, this would override previous user's shlink endpoint.